engine/updategraph/src/main/java/io/deephaven/engine/updategraph/UpdateGraphProcessor.java

```diff
@@ -1485,19 +1501,35 @@ private void refreshTablesAndFlushNotifications() {
             if (watchdogJob != null) {
                 sched.cancelJob(watchdogJob);
             }
-            final long cycleTime = System.nanoTime() - startTimeNanos;
-            if (cycleTime >= minimumCycleDurationToLogNanos) {
+            gcIntrospectionContext.endSample();
+            final long cycleTimeNanos = System.nanoTime() - startTimeNanos;
+            if (cycleTimeNanos >= minimumCycleDurationToLogNanos) {
                 if (suppressedCycles > 0) {
                     logSuppressedCycles();
                 }
-                log.info().append("Update Graph Processor cycleTime=").appendDouble(cycleTime / 1_000_000.0)
-                        .append("ms, lockWaitTime=").appendDouble(currentCycleLockWaitTotalNanos / 1_000_000.0)
-                        .append("ms, yieldTime=").appendDouble(currentCycleYieldTotalNanos / 1_000_000.0)
-                        .append("ms, sleepTime=").appendDouble(currentCycleSleepTotalNanos / 1_000_000.0)
-                        .append("ms").endl();
-            } else if (cycleTime > 0) {
-                suppressedCycles++;
-                suppressedCyclesTotalNanos += cycleTime;
+                final long gcTimeMillis = gcIntrospectionContext.deltaCollectionTimeMs();
+                final double cycleTimeMillis = cycleTimeNanos / 1_000_000.0;
+                LogEntry entry = log.info()
+                        .append("Update Graph Processor cycleTime=").appendPositiveDouble(cycleTimeMillis, 3)
+                        .append("ms, gcTime=").append(gcTimeMillis)
+                        .append("ms, gcTimePct=");
+                if (gcTimeMillis > 0 && cycleTimeMillis > 0.0) {
+                    final double gcTimePct = 100.0 * gcTimeMillis / cycleTimeMillis;
+                    entry = entry.appendPositiveDouble(gcTimePct, 2);
+                } else {
+                    entry = entry.append("0");
+                }
+                entry = entry.append("%, lockWaitTime=");
```

Comment:
After your if up there, I think I would go back to the standard chained append calls.

Reply:
We can't call appendAsMillisFromNanos as part of a chain.